mobile/src/common/index.tsx

```diff
@@ -0,0 +1 @@
+export { RoundedAvatar } from './rounded-avatar';
```

Comment:
We've discussed it already, having index file doesn't make development much easier but adds some problems with export, navigation and refactoring, so please do not use them.

Reply:
Yes, we'd discussed it previously, but I don't see any issues with the component usage/import in practice, so does it really happen? Could you please give an example?

Reply:
Sure.
Typically if you print, say, `Round...` the auto-complete suggests you to import something from the list of found literals. However, in that case there are two *different* literals that reside in different files, so it gives you a choice of two. It also gives you a problem with `go to definition` because it sends you to the index file instead of the source.
Sometimes it happens that some objects from the folder are imported from the index file whereas others are imported from specific files which adds to confusion.
For some autocomplete tools default option is to export it directly from the original file, developers have to go and change paths manually and people tend to forget to do that if there are no automatic tools that check that. There are many other small inconveniences, i won't describe all of them.
On top of that, using 'index' files can create some problems during compilation (it depends on `module` setting in tsconfig file. At the moment, react-native asks for the module type that doesn't suffer from that, but it might change in future version (although i doubt that :) ).

We've tried this approach on some older projects, faced that problems and decided to give it away, so we do not use it anymore.

Reply:
if we want to have Web version in the future, we will try to reuse some code parts, and we will probably use Webpack on client side. Webpack handles bad index.ts (and treeshaking probably won't work properly) :smile:

Reply:
ok, we already agreed do not use it, thanks for details